examples/msal-java/pom.xml

```diff
@@ -13,8 +13,8 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <maven.compiler.source>1.7</maven.compiler.source>
-    <maven.compiler.target>1.7</maven.compiler.target>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
```

Comment:
Changed minimum language level to 1.8 in order to allow a lambda expression in `Mono.defer`

Reply:
I don't think that's unreasonable given msal4j itself requires at least 1.8